PlasmaMVP/feeStructure.md

```diff
@@ -1,17 +1,35 @@
+We can require all participants of the side chain to allocate and bond a fee budget of Ether on the rootchain. In addition to the previous constraints on what constitutes a valid transaction, we impose that any accounts attempting to spend a transcation must specify a fee denomination in the txBytes that is checked against the rootchain. Per every transaction processed, the validator for a current round will deduct from all corresponding participant's fee budget. However, this opens up another order of complexity that we would like to avoid.  
+
+Does this change introduce a new challenge mechansim? Are there checks to ensure that the validator only deducts the fee amount as specified in the txBytes? How much code will this change/introduce in the rootchain?  
+
+  For these questions alone, this solution is not sufficient
 
-### Claim Fee:
-When a Plasma validator would like to withdraw his fees, he submits the amount he would like to withdraw, and his exit is added to the exitQueue with the lowest possible priority of the current block (currBlockNum, 2^16, 1). This way, users can verify that the amount he would like to withdraw doesn't exceed the amount of fees he has earned.
-* In the case of a valid fee withdrawal, his exit will not be challenged and his fee will be sent after the 1 week period.
-* In the case of an invalid fee withdrawal, users will see that and initiate a mass withdrawal. Since the validator has the lowest priority, all other users' exits will be processed before his, leaving him with the true fee amount.
 
 ### Fees Stored in Utxos
-Fees are included in the transaction. When a validator creates a block, he reserves a utxo at the last leaf (to give himself the lowest priority in the block) of the block where he can aggregate all the fees of this block. This utxo will look like a deposit utxo. Users can check that the amount stored in this fee utxo is correct. If the amount or transaction is found to be invalid, users will mass exit.
+In our earlier discussion of a plasma side chain vulnerability, [Ethereum Research Post](https://ethresear.ch/t/plasma-vulnerabiltity-sybil-txs-drained-contract/1654), regarding the plasma mvp spec not specifying the requirement of storing confirm sigs on-chain, we can exploit this to solve our issues with fees!  
+
+By requiring that a utxo cannot be spent without the correct confirm signatures and also including in them in the txBytes stored on-chain, spending a utxo on the child chain allows anyone watching to successfully challenge an invalid attempt to exit any grandparent of that particular utxo. We claim that that this requirement makes it very unlikely to exit a invalid uxto successfully.  
```

Comment:
I feel like we can cut this down a bit. The two conditionals in the first sentence are saying similar things. Requiring confirm sigs is very similar to including them on chain. Perhaps something like, "By requiring confirm signatures be included in a transaction, spending a utxo...challenge an invalid withdrawal of any earlier ancestor of that particular utxo"

Reply:
I agree